### DISCUSSION TITLE
Help Please

### DISCUSSION THREAD

**Discussion by openupshop** [awawa-dev/HyperHDR#255] [Q&A]
Hello Guys/Gals,

Hope everyone is having a great day & if not, hope your day turns into a great one.    I am using APA102 leds with a Raspberry PI 4 and I am accessing the PI through Putty.  So I got everything working but have a couple of issues.

1) I access the HYPERHDR GUI via windows browser, Pi_IP_Address:8090.  The GUI comes up, but I am unable to change anything, the "save" button in each subsection acts as if its "grayed out."  The "import" button acts the same way.  Nothing happens when I click on it.  I restarted my PI, enabled & disabled the leds, tried accessing the GUI from another device.

2) Before the issue above, I was also having an issue where the Ambilight would work great, but I would get intermittent flickering with random leds.  My data & Clock wires going from the leds to the PI's GPIO 19 & 23 respectively, are about 6 inches, so no long wires.  All my HDMI cables are new, 48gbps and only 1 meter max.

I would appreciate your help figuring things out... Thank you so much.

**Comment by awawa-dev** [comment]
Hi
1) Check if the system is read-only (advanced -> about)
2) If you are sure your installation then maybe it's time to think about the voltage level shifter (and common ground if there are separate power suppliers)?

**Reply by openupshop** [reply]
Thank you so much.... I totally forgot, I had followed a read only guide following installation and my dumb ass forgot that.  Thank you🙏🙏🙏
I do have separate power, but already have common ground.  Initially I did have a level shifter but was still getting intermittent, random flickering.  

I have also seen others mentioning adding a capacitor rated for more voltage than the power supply close to the first led _"to buffer sudden changes in current"_  Additionally, I've seen others add a resistor as well, although not sure what resistor and where.  Would any of these options help with the flickering?  And if so, any idea on what resistor and where to add it?

Thank you so much, much appreciate it.

**Comment by awawa-dev** [comment]
Which level shifter model did you use? I haven't use a capacitor in any of mine project but I guess it can protect the LEDs in certain situations: I prefer to buy a solid and branded power supply and so far it works. I had one damaged LED strip but it failed almost at the beginning after I completed the setup so I guess it came damaged: flickering with random colors starts always from one LED in the strip and it was easy to diagnose it and to replace that small segment.

**Comment by pimw1** [comment]
I advice to buy an quinled dig quad or quinled dig uno. And it's also good to read this blog that explains a lot about data signals: https://quinled.info/data-signal-cable-conditioning/. I did the same and all my flickering errors are gone,
